cmd/pool/sidecar.go

```diff
@@ -67,64 +59,107 @@ func sidecarOffer(ctx *cli.Context) error {
 	}
 
 	var (
-		args              = ctx.Args()
-		capacity, pushAmt uint64
-		duration          uint32
+		bid *poolrpc.Bid
+		err error
 	)
 
-	switch {
-	case ctx.IsSet("capacity"):
-		capacity = ctx.Uint64("capacity")
-	case args.Present():
-		parsed, err := parseAmt(args.First())
-		if err != nil {
-			return fmt.Errorf("unable to decode capacity: %v", err)
+	// If auto is set, then we'll only need to parse out a hand full of
+	// fields to submit a valid ticket.
+	if ctx.Bool("auto") {
+		var (
+			args              = ctx.Args()
+			capacity, pushAmt uint64
+			duration          uint32
+		)
+
+		switch {
+		case ctx.IsSet("capacity"):
+			capacity = ctx.Uint64("capacity")
+		case args.Present():
+			parsed, err := parseAmt(args.First())
+			if err != nil {
+				return fmt.Errorf("unable to decode capacity: %v", err)
+			}
+			capacity = uint64(parsed)
+			args = args.Tail()
 		}
-		capacity = uint64(parsed)
-		args = args.Tail()
-	}
 
-	switch {
-	case ctx.IsSet("self_chan_balance"):
-		pushAmt = ctx.Uint64("self_chan_balance")
-	case args.Present():
-		parsed, err := parseAmt(args.First())
-		if err != nil {
-			return fmt.Errorf("unable to decode self channel "+
-				"balance: %v", err)
+		switch {
+		case ctx.IsSet("self_chan_balance"):
+			pushAmt = ctx.Uint64("self_chan_balance")
+		case args.Present():
+			parsed, err := parseAmt(args.First())
+			if err != nil {
+				return fmt.Errorf("unable to decode self channel "+
+					"balance: %v", err)
+			}
+			pushAmt = uint64(parsed)
+			args = args.Tail()
+		}
+
+		switch {
+		case ctx.IsSet("lease_duration_blocks"):
+			duration = uint32(ctx.Uint64("lease_duration_blocks"))
+		case args.Present():
+			duration64, err := strconv.ParseInt(args.First(), 10, 32)
+			if err != nil {
+				return fmt.Errorf("unable to parse lease duration "+
+					"blocks: %v", err)
+			}
+			duration = uint32(duration64)
+			args = args.Tail()
 		}
-		pushAmt = uint64(parsed)
-		args = args.Tail()
-	}
 
-	switch {
-	case ctx.IsSet("lease_duration_blocks"):
-		duration = uint32(ctx.Uint64("lease_duration_blocks"))
-	case args.Present():
-		duration64, err := strconv.ParseInt(args.First(), 10, 32)
+		bid = &poolrpc.Bid{
+			Details: &poolrpc.Order{
+				Amt: capacity,
+			},
+			SelfChanBalance:     pushAmt,
+			LeaseDurationBlocks: duration,
+		}
+
+	} else {
+		// Otherwise, will parse out the full bid as normal.
+		bid, _, err = parseBaseBid(ctx)
 		if err != nil {
-			return fmt.Errorf("unable to parse lease duration "+
-				"blocks: %v", err)
+			return err
+
 		}
-		duration = uint32(duration64)
-		args = args.Tail()
 	}
 
 	client, cleanup, err := getClient(ctx)
 	if err != nil {
 		return err
 	}
+
 	defer cleanup()
 
+	// Give the user a chance to confirm the order details as this is
+	// binding once submitted.
+	if !ctx.Bool("force") {
+		if err := printOrderDetails(
+			client, btcutil.Amount(bid.Details.Amt),
+			order.SupplyUnit(bid.Details.MinUnitsMatch).ToSatoshis(),
```

Comment:
In the non-auto case this will result in an `integer divide by zero` error because the `MinUnitsMatch` is not set. Probably should only print the order details in the `auto` case.